app/models/organization_table_info.rb

```diff
@@ -0,0 +1,165 @@
+class OrganizationTableInfo
+  attr_accessor :organizations_counts, :total_counts
+
+  def initialize(org_stats, total_counts)
+    @organizations_counts = org_stats.kind_of?(Array) ? org_stats : [ org_stats ]
+    @total_counts = total_counts
+  end
+
+  def self.top_stats(top: 0, organization: nil, weeks: 5)
+    # Retrieve the results count, payloads count and organization info for the top most consuming organizations
+    org_payloads_and_results_count = Organization
+      .select('
+        organizations.id, organizations.api_id, organizations.name, organizations.display_name,
+        SUM(test_payloads.results_count) AS test_results_count, COUNT(test_payloads.id) as test_payloads_count
+      ')
+      .joins('
+        LEFT OUTER JOIN "projects" ON "projects"."organization_id" = "organizations"."id"
+        LEFT OUTER JOIN "project_versions" ON "project_versions"."project_id" = "projects"."id"
+        LEFT OUTER JOIN"test_payloads" ON "test_payloads"."project_version_id" = "project_versions"."id"
+      ')
+      .group('organizations.id')
+      .order('test_results_count DESC NULLS LAST')
```

Comment:
For the rare but possible cases where organizations have the same results count (or no results), I would add a second order criteria on the name to have predictable ordering.